### PR TITLE
refactor: Refactor git_branch module to use new module config

### PR DESCRIPTION
**Pull request by matiaskotlik**
#### Description
Refactored git_branch module to use new module config. The documentation for the module was also incorrect, a value that was supposed to be an integer was a string in the example, so I fixed that too after it tripped me up when I was testing the module.

#### Motivation and Context

Related #383 

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: src/configs/git_branch.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+use ansi_term::{Color, Style};
+use starship_module_config_derive::ModuleConfig;
+
+#[derive(Clone, ModuleConfig)]
+pub struct GitBranchConfig<'a> {
+    pub symbol: SegmentConfig<'a>,
+    pub truncation_length: i64,
+    pub truncation_symbol: &'a str,
+    pub branch_name: SegmentConfig<'a>,
```
Comment (matiaskotlik) [inline src/configs/git_branch.rs]: This segment was named `name` in the original file, but I thought it was a little ambiguous if it is going to be customizeable from the config file. Should I change it back? Or should we go all the way and even change the name of the segment (line 42 of modules/git_branch.rs) to "branch_name"?

Comment (heyrict) [reply src/configs/git_branch.rs]: Ok for both as these configs are currently undocumented and are not officially available for users. I've opened an issue for that #472.

File: src/modules/git_branch.rs [hunk 2]
```diff
@@ -1,43 +1,34 @@
...
     // TODO: Once error handling is implemented, warn the user if their config
     // truncation length is nonsensical
-    let len = if unsafe_truncation_length <= 0 {
-        log::debug!(
-            "[WARN]: \"truncation_length\" should be a positive value, found {}",
-            unsafe_truncation_length
+    let len = if config.truncation_length <= 0 {
+        log::warn!(
+            "\"truncation_length\" should be a positive value, found {}",
+            config.truncation_length
```
[the remark this answers is not in the document]
Comment (matiaskotlik) [reply src/modules/git_branch.rs]: I don't really understand what's going on #41 haha. When I first saw the code it seemed like the `warn!` macro didn't exist or something. Should I undo the change then?

[the remark this answers is not in the document]
Comment (matchai) [reply src/modules/git_branch.rs]: I'm in favor of using the `warn!` macro instead of logging a `[WARN]` prefix. 👍

**Pull request merged**